### PR TITLE
[#4171] feat(docs): Add Paimon Catalog doc

### PR DESCRIPTION
**Pull request by caican00**
### What changes were proposed in this pull request?
Add Paimon Catalog doc

### Why are the changes needed?
Fix: https://github.com/apache/gravitino/issues/4171

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
No testing required.

### CODE REVIEW

File: docs/lakehouse-paimon-catalog.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+:::
+
+## Catalog
+
+### Catalog capabilities
+
+- Works as a catalog proxy, supporting `FilesystemCatalog`.
+- Supports DDL operations for Paimon schemas and tables.
+
+- Doesn't support `JdbcCatalog` and `HiveCatalog` backend-catalog now.
```
Comment (FANNG1) [inline docs/lakehouse-paimon-catalog.md:31]: backend-catalog -> catalog backend?

File: docs/lakehouse-paimon-catalog.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+- Doesn't support alterTable now.
```
Comment (FANNG1) [inline docs/lakehouse-paimon-catalog.md:32]: add doesn't support partition, distribution, sortorder?

File: docs/lakehouse-paimon-catalog.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+| `warehouse`                                        | Warehouse directory of catalog. `file:///user/hive/warehouse-paimon/` for local fs or `hdfs://namespace/hdfs/path` for HDFS.                                                                                | (none)                 | Yes                                                             | 0.6.0         |
+| `authentication.type`                              | The type of authentication for Paimon catalog backend, currently Gravitino only supports `Kerberos` and `simple`.                                                                                           | `simple`               | No                                                              | 0.6.0         |
+| `authentication.kerberos.principal`                | The principal of the Kerberos authentication                                                                                                                                                                | (none)                 | required if the value of `authentication.type` is Kerberos.     | 0.6.0         |
```
Comment (FANNG1) [inline docs/lakehouse-paimon-catalog.md:43]: missing `.` for `The principal of the Kerberos authentication`

File: docs/lakehouse-paimon-catalog.md (hunk 2 above)
Comment (caican00) [reply docs/lakehouse-paimon-catalog.md:32]: i added them here.
https://github.com/apache/gravitino/pull/4192/files#diff-b601aacad7f88ce976fdfaa7008127fe0f3241023715eba626ff45998a5a61bbR86-R100

File: docs/lakehouse-paimon-catalog.md (hunk 1 above)
Comment (caican00) [reply docs/lakehouse-paimon-catalog.md]: updated.

Comment (caican00) [reply docs/lakehouse-paimon-catalog.md]: updated.

**Comment by caican00** [comment]
comments have been addressed. cc @FANNG1

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
merged to main, thanks for your contribution!

**Pull request merged**